cpp/src/arrow/meson.build

```diff
@@ -264,6 +264,79 @@ else
     rapidjson_dep = disabler()
 endif
 
+azure_dep = disabler()
+gcs_dep = disabler()
+s3_dep = disabler()
+if needs_filesystem
+    arrow_filesystem_srcs = [
+        'filesystem/filesystem.cc',
+        'filesystem/localfs.cc',
+        'filesystem/mockfs.cc',
+        'filesystem/path_util.cc',
+        'filesystem/util_internal.cc',
+    ]
+
+    arrow_filesystem_deps = []
+
+    if needs_azure
+        arrow_filesystem_srcs += ['filesystem/azurefs.cc']
+        cmake = import('cmake')
+        azure_opt = cmake.subproject_options()
+        azure_opt.add_cmake_defines(
+            {'BUILD_PERFORMANCE_TESTS': 'FALSE'},
+            {'BUILD_SAMPLES': 'FALSE'},
+            {'BUILD_TESTING': 'FALSE'},
+            {'BUILD_WINDOWS_UWP': 'TRUE'},
+            {'CMAKE_UNITY_BUILD': 'FALSE'},
+            {'DISABLE_AZURE_CORE_OPENTELEMETRY': 'TRUE'},
+            # TODO: couldn't find any usage of this in Arrow - do we need?
+            # set(ENV{AZURE_SDK_DISABLE_AUTO_VCPKG} TRUE)
```

Comment:
Azure SDK for C++ prepared vcpkg automatically without this: https://github.com/Azure/azure-sdk-for-cpp/blob/27c2e5d335c9306d08aa49bd5fe92d0cc1628fe0/cmake-modules/AzureVcpkg.cmake#L17-L41

We don't want to use vcpkg here.

Reply:
Ok great - I will set to FALSE then. Should I update the CMake configuration as well?

Reply:
Ah nevermind - ignore comment above. I was thinking this ENABLED something but I see it DISABLEs